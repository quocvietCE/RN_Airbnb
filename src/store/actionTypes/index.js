export const AUTH_ACTION = {
  USER_LOGIN: 'USER_LOGIN',
  USER_LOGIN_SUCCESS: 'USER_LOGIN_SUCCESS',
  USER_LOGIN_FAILURE: 'USER_LOGIN_FAILURE',

  LOG_OUT: 'LOG_OUT',
};

export const APP_ACTION = {
  UPDATE_LANGUAGE: 'UPDATE_LANGUAGE',
  REQUIRE_LOGIN_PAGE: 'REQUIRE_LOGIN_PAGE',
  UPDATE_POSITION: 'UPDATE_POSITION',
  OPEN_MODAL: 'OPEN_MODAL',
  CLOSE_MODAL: 'CLOSE_MODAL',
  OPEN_ADDRESS_MODAL: 'OPEN_ADDRESS_MODAL',
  CLOSE_ADDRESS_MODAL: 'CLOSE_ADDRESS_MODAL',
  CHANGED_COUNTRY: 'CHANGED_COUNTRY',
  CHANGE_COUNTRY_SUCCESS: 'CHANGE_COUNTRY_SUCCESS',
  SET_APP_DATA: 'SET_APP_DATA',
  GET_CONFIGURATION_APP: 'GET_CONFIGURATION_APP',
  GET_CONFIGURATION_APP_SUCCESS: 'GET_CONFIGURATION_APP_SUCCESS',
  GET_CONFIGURATION_APP_FAIL: 'GET_CONFIGURATION_APP_FAIL',
  CLEAR_ERROR: 'CLEAR_ERROR',
  OPEN_FULL_MAP_VIEW: 'OPEN_FULL_MAP_VIEW',
  CLOSE_FULL_MAP_VIEW: 'CLOSE_FULL_MAP_VIEW',
  SAVE_PERMISSION_PHOTO: 'SAVE_PERMISSION_PHOTO',
};

export const LISTING_ACTION = {
  GET_LISTING_DATA: 'GET_LISTING_DATA',
  GET_LISTING_DATA_SUCCESS: 'GET_LISTING_DATA_SUCCESS',
  GET_LISTING_DATA_FAILURE: 'GET_LISTING_DATA_FAILURE',
};

export const NETWORK_ACTION = {
  SET_STATUS_CONNECTED: 'SET_STATUS_CONNECTED',
  SET_STATUS_CONNECTED_SUCCESS: 'SET_STATUS_CONNECTED_SUCCESS',
  SET_STATUS_CONNECTED_FAILURE: 'SET_STATUS_CONNECTED_FAILURE',
};
