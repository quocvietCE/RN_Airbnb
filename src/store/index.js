import {createStore, applyMiddleware, compose} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {persistStore, persistReducer} from 'redux-persist';
import AsyncStorage from '@react-native-community/async-storage';
import logger from 'redux-logger';
import rootSaga from './sagas';

import rootReducer from './reducers';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  // blacklist: ['auth', 'app'],
  blacklist: ['app, listing'],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);
const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const sagaMiddleware = createSagaMiddleware();

export const store = createStore(
  persistedReducer,
  composeEnhancer(applyMiddleware(sagaMiddleware, logger)),
);

export const persistor = persistStore(store);

sagaMiddleware.run(rootSaga);
export default () => ({store, persistor});
