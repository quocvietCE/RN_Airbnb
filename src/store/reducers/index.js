import {combineReducers} from 'redux';
import auth from './authReducer';
import app from './appReducer';
import listing from './listingReducer';
import network from './networkReducer';

export default combineReducers({
  auth,
  app,
  listing,
  network,
});
