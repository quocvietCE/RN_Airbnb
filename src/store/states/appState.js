const initState = {
  countryCode: null,
  languageCode: 'en',
  countryIndex: null,
  callingCode: null,
  errorMessage: null,
  isLoading: false,
  pageRequired: null,
  modalStatus: null,
  modalData: null,
  openAddressModal: false,
  configs: {},

  permission: true,
};

const setRequestSuccess = () => ({
  errorMessage: null,
  isLoading: false,
});

const setRequestFailed = (message) => ({
  errorMessage: message,
  isLoading: false,
});

const setPageRequiredName = (pageName) => ({
  pageRequired: pageName,
});

const closeModal = () => ({
  modalStatus: null,
  modalData: null,
});

const appState = {
  initState,
  setRequestSuccess,
  setRequestFailed,
  setPageRequiredName,
  closeModal,
};

export default appState;
