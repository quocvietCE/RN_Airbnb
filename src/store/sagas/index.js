import {takeLatest, all, takeEvery} from 'redux-saga/effects';
import authSaga from './authSaga';
import {AUTH_ACTION} from '../actionTypes';

function* handleLogin() {
  yield takeLatest(AUTH_ACTION.USER_LOGIN, authSaga.handleLogin);
}

export default function* rootSaga() {
  yield all([handleLogin()]);
}
