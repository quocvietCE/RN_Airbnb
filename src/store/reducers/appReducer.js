import {APP_ACTION, AUTH_ACTION, LISTING_ACTION} from '../actionTypes';
import appState from '../states/appState';

const authReducer = (state = appState.initState, action) => {
  switch (action.type) {
    case AUTH_ACTION.USER_LOGIN:
    case LISTING_ACTION.GET_LISTING_DATA:
      return {
        ...state,
        isLoading: true,
      };
    case AUTH_ACTION.USER_LOGIN_SUCCESS:
    case LISTING_ACTION.GET_LISTING_DATA_SUCCESS:
      return {
        ...state,
        ...appState.setRequestSuccess(),
      };
    case AUTH_ACTION.USER_LOGIN_FAILURE:
    case LISTING_ACTION.GET_LISTING_DATA_FAILURE:
      return {
        ...state,
        ...appState.setRequestFailed(action.errorObject),
      };
    case APP_ACTION.CLEAR_ERROR:
      return {
        ...state,
        errorObject: {},
      };
    default:
      return state;
  }
};

export default authReducer;
