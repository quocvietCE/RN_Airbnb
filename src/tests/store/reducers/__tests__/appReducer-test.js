import 'react-native';
import appReducer from '../../../../store/reducers/appReducer';
import appState from '../../../../store/states/appState';
import {APP_ACTION, AUTH_ACTION} from '../../../../store/actionTypes';

describe('Test appReducer', () => {
  it('returns the same appReducer state on an unhandled state', () => {
    expect(appReducer(appState.initState, {type: '_NULL'})).toMatchSnapshot();
  });

  it('Check AUTH_ACTION.USER_LOGIN_FAILURE type', () => {
    expect(
      appReducer(appState.initState, {
        type: AUTH_ACTION.USER_LOGIN,
      }),
    ).toEqual({
      ...appState.initState,
      isLoading: true,
    });
  });

  it('Check Snapshot AUTH_ACTION.USER_LOGIN_SUCCESS type', () => {
    expect(
      appReducer(appState.initState, {
        type: AUTH_ACTION.USER_LOGIN_SUCCESS,
      }),
    ).toMatchSnapshot();
  });

  it('Check Snapshot AUTH_ACTION.USER_LOGIN_FAILURE type', () => {
    expect(
      appReducer(appState.initState, {
        type: AUTH_ACTION.USER_LOGIN_FAILURE,
        error: 'Network Error',
      }),
    ).toMatchSnapshot();
  });

  it('Check AUTH_ACTION.USER_LOGIN_FAILURE type', () => {
    expect(
      appReducer(appState.initState, {
        type: AUTH_ACTION.USER_LOGIN_FAILURE,
        error: 'Network Error',
      }),
    ).toEqual({
      ...appState.initState,
      errorMessage:
        'Sorry, it looks like the Username and/or Password you provided does not match our records',
      isLoading: false,
    });
  });

  it('Check Snapshot APP_ACTION.CLEAR_ERROR type', () => {
    expect(
      appReducer(appState.initState, {
        type: APP_ACTION.CLEAR_ERROR,
      }),
    ).toMatchSnapshot();
  });
});
