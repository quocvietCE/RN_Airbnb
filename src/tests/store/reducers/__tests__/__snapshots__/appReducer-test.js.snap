// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test appReducer Check AUTH_ACTION.USER_LOGIN_FAILURE type 1`] = `
Object {
  "errorMessage": "Network Error",
  "isLoading": false,
  "languageCode": "en",
}
`;

exports[`Test appReducer Check Snapshot APP_ACTION.CLEAR_ERROR type 1`] = `
Object {
  "errorObject": Object {},
  "isLoading": false,
  "languageCode": "en",
  "objectError": Object {},
}
`;

exports[`Test appReducer Check Snapshot AUTH_ACTION.USER_LOGIN_FAILURE type 1`] = `
Object {
  "errorObject": undefined,
  "isLoading": false,
  "languageCode": "en",
}
`;

exports[`Test appReducer Check Snapshot AUTH_ACTION.USER_LOGIN_SUCCESS type 1`] = `
Object {
  "errorObject": Object {},
  "isLoading": false,
  "languageCode": "en",
}
`;

exports[`Test appReducer returns the same appReducer state on an unhandled state 1`] = `
Object {
  "errorObject": Object {},
  "isLoading": false,
  "languageCode": "en",
}
`;
